src/lib/index/publish.js

```diff
@@ -27,25 +27,77 @@
 'use strict';
 
 const
-	Amqp = require('./shared/amqp'),
+	_ = require('lodash'),
+	amqp = require('amqplib'),
+	{ validate } = require('./shared/configValidator'),
 
 	EventEmitter = require('events'),
 
 	ERROR_EVENT = 'error_event',
 
 	emitter = new EventEmitter(),
 
-	send = ({ eventName, buffer, config }) => {
-		return Amqp.apply(channel => channel.sendToQueue(eventName, buffer), config)
-			.catch(err => {
-				emitter.emit(ERROR_EVENT, err.message);
-				throw err;
-			});
-	};
+	publishers = new Map();
+
+class Publisher {
+
+	constructor(config) {
+		validate(config);
+		this.config = _.cloneDeep(config);
+	}
+
+	async init() {
+		this.connection = await amqp.connect(this.config.cloudamqpUrl);
+		this.channel = await this.connection.createChannel();
+	}
+
+	async publish(queue, buffer) {
+		return await this.channel.sendToQueue(queue, buffer);
+	}
+
+	async close() {
+		await this.connection.close();
```

Comment:
Are you sure you want to return undefined? Why not return this.connection.close()?

Reply:
Good point. The caller might want to wait until everything is closed.

Reply:
They already can as it stands. They just don't have access to the return value of close()... if indeed there is one.

Reply:
True